File: Codechef-2019/JUN19B/CHFING.py
# -*- coding: utf-8 -*-
"""
@author: anishukla
"""

""" Link to the problem: 
    https://www.codechef.com/JUNE19B/problems/CHFING
    Link to my profile:
    https://www.codechef.com/users/an1shukla """

T = int(input())
ref = 1000000007

for i in range(T):
    N, K = input().split()
    N = int(N)
    K = int(K)
    req = N-1
    end = K-1
    start = end%req
    val = ((end-start)//req) + 1
    add = start + end
    if(val%2 == 0):
        val = val//2
        
    else:
        add = add//2
        
    op = 0
    while(add>0):
        if(add%2 == 1):
            op = op + val
        
        op = op%ref
        val = val*2
        val = val%ref
        
        add = add//2
    
    op = op%ref
        
    print(op%ref)

File: Codechef-2019/JUN19B/CHFING.cpp
/*
Link to the problem:
https://www.codechef.com/JUNE19B/problems/CHFING/
Link to my profile:
https://www.codechef.com/users/purbasa_kar
*/
#include<iostream>
using namespace std;
int main()
{
    long long int T,K,N,m,v,a,ans;
    cin>>T;
    while(T>0)
    {
        cin>>N>>K;
        ans=0;
        m=(K-1)%(N-1);
        v=(((K-1)-m)/(N-1))+1;
        a=m+(K-1);
        if(v%2==1)
            a=a/2;
        else
            v=v/2;
        while(a>0)
        {
            if(a%2==1)
                ans=ans+v;
            v=v*2;
            v=v%1000000007;
            a=a/2;
            a=a%1000000007;
        }
        cout<<ans%1000000007<<endl;
        T--;
    }
    return 0;
}
